### DISCUSSION TITLE
Attach schema to JSON field and validate data by it on save

### DISCUSSION THREAD

**Discussion by MurzNN** [keystonejs/keystone#6372] [Event Q&A Q3 2021]
JSON fields are cool, but they are too nonrestrictive! So there are big changes to save non-valid JSON. To solve this problem will be good to implement [JSON Schema](https://json-schema.org/) support (attaching schema to field) and validation by it on backend. Yes, we can make this via custom code, but will be good to have it in core.

**Comment by bladey** [comment]
Thanks for the feature suggestion @MurzNN!

**Comment by MurzNN** [comment]
Having schema, we even can improve rendering of edit form in Admin UI, to convert simple plaintext textarea to multiple inputs of several types (string, number, array of strings, etc).

**Comment by raveling** [comment]
@MurzNN we provided an answer to this question in the recent Q&A event:
https://www.youtube.com/watch?v=r1IJh-iHm1c&t=1085s
